pandas/io/json/json.py

```diff
@@ -320,49 +334,140 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
                 {"index": "row 2", "col 1": "c", "col 2": "d"}]}'
     """
 
+    if chunksize is not None:
+        chunksize = _validate_integer("chunksize", chunksize, 1)
+        if not lines:
+            raise ValueError("chunksize should only be passed if lines=True")
+
     filepath_or_buffer, _, _ = get_filepath_or_buffer(path_or_buf,
                                                       encoding=encoding)
-    if isinstance(filepath_or_buffer, compat.string_types):
-        try:
-            exists = os.path.exists(filepath_or_buffer)
-
-        # if the filepath is too long will raise here
-        # 5874
-        except (TypeError, ValueError):
-            exists = False
-
-        if exists:
-            fh, handles = _get_handle(filepath_or_buffer, 'r',
-                                      encoding=encoding)
-            json = fh.read()
-            fh.close()
-        else:
-            json = filepath_or_buffer
-    elif hasattr(filepath_or_buffer, 'read'):
-        json = filepath_or_buffer.read()
+
+    json_reader = JsonReader(
+        filepath_or_buffer, orient=orient, typ=typ, dtype=dtype,
+        convert_axes=convert_axes, convert_dates=convert_dates,
+        keep_default_dates=keep_default_dates, numpy=numpy,
+        precise_float=precise_float, date_unit=date_unit, encoding=encoding,
+        lines=lines, chunksize=chunksize
+    )
+
+    if chunksize:
+        return json_reader
+
     else:
-        json = filepath_or_buffer
+        return json_reader.read()
 
-    if lines:
+
+class JsonReader(BaseIterator):
+    """
+    Reads a JSON document to a pandas object.
+
+    If initialized with ``lines=True`` and ``chunksize``, can be iterated over
+    ``chunksize`` lines at a time.
+    """
+    def __init__(
+        self, filepath_or_buffer, orient, typ, dtype, convert_axes,
+        convert_dates, keep_default_dates, numpy, precise_float, date_unit,
+        encoding, lines, chunksize, raw_json=False
+    ):
```

Comment:
I think we can move this parenthesis up to the previous line.  This looks more Java than Python :smile:

Reply:
done